### PR TITLE
fix(Dimmer): fade background color on open/close

### PR DESCRIPTION
**Pull request by matebek-tw**
## 🖼 Context

https://transferwise.atlassian.net/jira/software/projects/DS/boards/277?label=Web&selectedIssue=DS-1212

## 🚀 Changes

- Migrate `Dimmer` to TypeScript
- Move background color on `Dimmer` to the `.dimmer--enter-done` class

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/dimmer/Dimmer.tsx [hunk 1]
```diff
@@ -14,8 +13,19 @@ export const EXIT_ANIMATION = 350;
 
 const dimmerManager = new DimmerManager();
 
-export const handleTouchMove = (event) => {
-  const isTouchedElementDimmer = event.target.classList.contains('dimmer');
+type DimmerProps = PropsWithChildren<{
+  className?: string;
```
Comment (dozortsev) [inline packages/components/src/dimmer/Dimmer.tsx:17]: `className` should come from `CommonProps`

File: packages/components/src/dimmer/Dimmer.tsx [hunk 2]
```diff
@@ -118,36 +148,11 @@ export const DimmerContentWrapper = ({ children, scrollBody }) => {
...
-  disableClickToClose: false,
-  fadeContentOnEnter: false,
-  fadeContentOnExit: false,
-  transparent: false,
-  onClose: undefined,
-  open: false,
-  scrollable: false,
-};
-
 export { Dimmer };
```
Comment (dozortsev) [inline packages/components/src/dimmer/Dimmer.tsx:156]: I think you can now delete this statement ([for context here is similar example](https://github.com/transferwise/neptune-web/blob/main/packages/components/src/moneyInput/MoneyInput.js#L381-L383))

File: packages/components/src/dimmer/Dimmer.tsx [hunk 3]
```diff
@@ -118,36 +148,11 @@ export const DimmerContentWrapper = ({ children, scrollBody }) => {
...
 
 export default withNextPortal(Dimmer);
```
Comment (dozortsev) [inline packages/components/src/dimmer/Dimmer.tsx:158]: I think in order for tsc to properly generate correct types [this function](https://github.com/transferwise/neptune-web/blob/main/packages/components/src/withNextPortal/withNextPortal.js) needs written also on TS

**Review by dozortsev — COMMENTED** [review]
Looks good 👍 

But please make changes to improve types.

atm it generates following

```js
// at the end of Dimmer.d.ts file
..
export { Dimmer };
declare const _default: (props: any) => import('react').ReactPortal | null;
export default _default;
```

this doesn't look right 🤔 I think consumers will get following error/warning on their side, but I could be wrong

> JSX element type 'Dimmer' does not have any construct or call signatures

**Review by dozortsev — COMMENTED** [review]
I would also delete Dimmer from here https://transferwise.github.io/neptune-web/blog/2021-04-30-april-release#missing-types

**Review by dozortsev — APPROVED** [review]

File: packages/components/src/dimmer/Dimmer.tsx [hunk 4]
```diff
@@ -14,8 +13,19 @@ export const EXIT_ANIMATION = 350;
 
 const dimmerManager = new DimmerManager();
 
-export const handleTouchMove = (event) => {
-  const isTouchedElementDimmer = event.target.classList.contains('dimmer');
+type DimmerProps = CommonProps & {
+  children?: ReactNode;
```
Comment (dozortsev) [inline packages/components/src/dimmer/Dimmer.tsx:17]: maybe `React.PropsWithChidren`?

**Review by dozortsev — APPROVED** [review]

File: packages/components/src/withNextPortal/withNextPortal.tsx [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, ComponentType } from 'react';
 import { createPortal } from 'react-dom';
 
-export default function withNextPortalWrapper(Component) {
-  return function (props) {
+export default function withNextPortalWrapper<T>(Component: ComponentType<T>) {
```
Comment (dozortsev) [inline packages/components/src/withNextPortal/withNextPortal.tsx:4]: what type of data this function returns? shouldn't we explicitly define type here? 🤔

Comment (matebek-tw) [reply packages/components/src/withNextPortal/withNextPortal.tsx:4]: It either returns `ComponentType<T>` or `null`, TypeScript can assert the type based on the return value.

File: packages/components/src/dimmer/Dimmer.tsx (hunk 4 above)
Comment (dozortsev) [reply packages/components/src/dimmer/Dimmer.tsx:17]: ok

File: packages/components/src/withNextPortal/withNextPortal.tsx (hunk 5 above)
Comment (dozortsev) [reply packages/components/src/withNextPortal/withNextPortal.tsx:4]: ok

**Pull request merged**